comps/lvms/src/integrations/llava.py

```diff
@@ -112,11 +113,13 @@ async def invoke(
 
         inputs = {"img_b64_str": img_b64_str, "prompt": prompt, "max_new_tokens": max_new_tokens}
         # forward to the LLaVA server
-        response = requests.post(url=f"{self.base_url}/generate", data=json.dumps(inputs), proxies={"http": None})
+        async with aiohttp.ClientSession() as session:
```

Comment:
Think it will be more efficient if we initialize this session in __init__ and here we edit it as `async with self.session.post...`!

Reply:
@Spycsh From Kubernetes(K8s) perspective, I don't think make the session persistent is a good idea. I assume a persistent session will result in a persistent TCP connection. If so,  it will prevent the out-of-box K8s to do the traffic load-balancing if we have multiple same inference pods for this lvm_llava service to talk to. All the requests will be routed to a single inference pod, and other pods will sit idle all the time. Unless you add some client-side load-balancing logic in your logic here, or we have some k8s cluster admin to deploy/configure service meshes.

I would suggest to keep shot lived sessions to make leverage the K8s's out-of-box load balancing, unless we're sure the TCP connection time is the main reason of performance bottle neck.

Reply:
@lianhao if that really sticks k8s load-balancing yes we can keep the short-lived session as the code is! Better if we can  further look into the "stick" issue and if that is the case, we can have a flag to control whether to use diff approaches and leave room to reusing TCP connection on single node for optimized perf.